core/src/main/java/com/sequenceiq/cloudbreak/structuredevent/db/LegacyStructuredEventRepository.java

```diff
@@ -55,4 +56,13 @@ default Optional<StructuredEventEntity> findByNameAndWorkspaceId(String name, Lo
     default Set<StructuredEventEntity> findByNameInAndWorkspaceId(Set<String> name, Long workspaceId) {
         throw new UnsupportedOperationException();
     }
+
+    @Modifying
+    @Query(value = "DELETE FROM structuredevent se WHERE se.resourceId = :resourceId AND se.timestamp < :timestamp", nativeQuery = true)
+    void deleteRecordsOlderThan(@Param("resourceId") Long resourceid, @Param("timestamp") Long timestamp);
+
+    @Modifying
+    @Query(value = "DELETE FROM structuredevent se WHERE se.resourceCrn LIKE :resourceCrn AND se.timestamp < :timestamp", nativeQuery = true)
```

Comment:
Why do we need `LIKE` for filtering certain resource CRNs?

Reply:
because we are only interested in the account which is a part of the CRN. Since the table does not store the account itself, I was only able to figure out which resource belongs to a certain account by doing a LIKE

Reply:
`se.resourceCrn LIKE :resourceCrn` so it doesn't seem to correct, if we checking crn equality then we can avoid the expensive `LIKE` if it is an account id check then please update the name of the query parameter and let's use `LIKE`.